test/gnosisSafePersonalEdition.js

```diff
@@ -49,10 +51,37 @@ contract('GnosisSafePersonalEdition', function(accounts) {
         let executorDiff = await web3.eth.getBalance(executor) - executorBalance
         console.log("    Executor earned " + web3.fromWei(executorDiff, 'ether') + " ETH")
         assert.ok(executorDiff > 0)
-    })
+    });
+
+    it('should deposit and withdraw 1 ETH paying with token', async () => {
+        let token = await safeUtils.deployToken(accounts[0]);
+        let executorBalance = (await token.balances(executor)).toNumber();
+        await token.transfer(gnosisSafe.address, 10000000, {from: accounts[0]});
+        await web3.eth.sendTransaction({from: accounts[0], to: gnosisSafe.address, value: web3.toWei(1.1, 'ether')})
+        await safeUtils.executeTransaction(lw, gnosisSafe, 'executeTransaction withdraw 0.5 ETH', [lw.accounts[0], lw.accounts[2]], accounts[0], web3.toWei(0.5, 'ether'), "0x", CALL, executor, {
+          gasToken: token.address
+        });
+
+        await safeUtils.executeTransaction(lw, gnosisSafe, 'executeTransaction withdraw 0.5 ETH', [lw.accounts[0], lw.accounts[2]], accounts[0], web3.toWei(0.5, 'ether'), "0x", CALL, executor, {
+          gasToken: token.address
+        })
+        let executorDiff = (await token.balances(executor)).toNumber() - executorBalance;
+        console.log("    Executor earned " + executorDiff + " Tokens")
+        assert.ok(executorDiff > 0);
+    });
+
+    it('should fail when depositing 1 ETH paying with token due to token transfer fail', async () => {
```

Comment:
@rmeissner This should fail. But it doesnt

Reply:
That is because the final costs are probably not `91861`. Therefore the mock returns `0` which for us indicates a success (as expected by the original ERC20 token standard. If you look at the `transfer` method of https://etherscan.io/address/0xd26114cd6EE289AccF82350c8d8487fedB8A0C07#code you see that nothing is returned)

To mock everything we need to update the mock contract. Also it seems like the npm inclusion doesn't work on a fresh setup. You can include the MockContract.sol in our project again until it is updated ;)

Adjusted version of the mock contract:
```
pragma solidity ^0.4.23;

contract MockContract {
	enum MockType { Return, Revert, OutOfGas }
	mapping(bytes => MockType) mockTypes;
	mapping(bytes => bytes) expectations;
	mapping(bytes4 => MockType) mockTypesAny;
	mapping(bytes4 => bytes) expectationsAny;

	/**
	 * @dev Stores a response that the contract will return if the fallback function is called with the given method name and matching arguments.
	 * @param call ABI encoded calldata that if invoked on this contract will return `response`. Parameter values need to match exactly.
	 * @param response ABI encoded response that will be returned if this contract is invoked with `call`
	 */
	function givenReturn(bytes call, bytes response) public {
		mockTypes[call] = MockType.Return;
		expectations[call] = response;
	}

	function givenReturnAny(bytes4 method, bytes response) public {
		mockTypesAny[method] = MockType.Return;
		expectationsAny[method] = response;
	}

	function givenRevert(bytes call) public {
		mockTypes[call] = MockType.Revert;
	}

	function givenRevertAny(bytes4 method) public {
		mockTypesAny[method] = MockType.Revert;
	}

	function givenOutOfGas(bytes call) public {
		mockTypes[call] = MockType.OutOfGas;
	}

	function givenOutOfGasAny(bytes4 method) public {
		mockTypesAny[method] = MockType.OutOfGas;
	}

	function() payable public {
		bytes4 methodId;
		assembly {
			methodId := calldataload(0)
		}
		if (mockTypesAny[methodId] == MockType.Revert||mockTypes[msg.data] == MockType.Revert) {
			revert();
		}
		if (mockTypesAny[methodId] == MockType.OutOfGas||mockTypes[msg.data] == MockType.OutOfGas) {
			while(true) {
				assembly {
					sstore(sload(0x40), 1)
				}
			}
		}
		bytes memory result = expectationsAny[methodId];
		if (result.length == 0) {
			result = expectations[msg.data];
		}
		assembly {
			return(add(0x20, result), mload(result))
		}
	}
}
```

and mock with `await mockContract.givenRevertAny(transferData.slice(0, 10));`

cc @fleupold

Reply:
Thanks, let's update the mock contract and publish a new npm version quickly.

Maybe slightly simpler to use the new method would be:
```
var Web3EthAbi = require('web3-eth-abi');
const method = Web3EthAbi.encodeFunctionSignature('transfer(address,uint)')
await mockContract.givenRevertAny(method)
```